### PR TITLE
[fix-15907] Fix get remote shell exit code is incorrect

### LINKED ISSUE #15907: [Bug] [worker] The RemoteShellTask `getTaskExitCode` method may encounter situations where it cannot retrieve the exit code properly.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

The RemoteShellTask `getTaskExitCode` method may encounter situations where it cannot retrieve the exit code properly.

For example, in the following code, when curl returns a value, the exit code may not be parsed correctly.
image
image
image

### What you expected to happen

see above

### How to reproduce

see above

### Anything else

No response

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Zzih**
## Purpose of the pull request

fix https://github.com/apache/dolphinscheduler/issues/15907

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-task-plugin/dolphinscheduler-task-remoteshell/src/main/java/org/apache/dolphinscheduler/plugin/task/remoteshell/RemoteExecutor.java [hunk 1]
```diff
@@ -131,7 +131,9 @@ public Integer getTaskExitCode(String taskId) throws IOException {
         int exitCode = -1;
         log.info("Remote shell task run status: {}", logLine);
         if (logLine.contains(STATUS_TAG_MESSAGE)) {
-            String status = logLine.replace(STATUS_TAG_MESSAGE, "").trim();
+            int startIndex = logLine.indexOf(STATUS_TAG_MESSAGE) + STATUS_TAG_MESSAGE.length();
+            int endIndex = logLine.length();
+            String status = logLine.substring(startIndex, endIndex);
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-remoteshell/src/main/java/org/apache/dolphinscheduler/plugin/task/remoteshell/RemoteExecutor.java]:
```suggestion
            String status = StringUtils.substringAfter(logLine, STATUS_TAG_MESSAGE);
```

**Review by ruanwenjun — CHANGES_REQUESTED** [review]
Please directly use StringUtils

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by rickchengx — APPROVED** [review]
LGTM

**Comment by stoplyy** [comment]
It seems a new bug has been introduced. 

The .trim() in the original code is necessary, otherwise the case of xxx-0\n cannot be handled.

image

**Comment by saowu** [comment]
> It seems a new bug has been introduced.
> 
> The .trim() in the original code is necessary, otherwise the case of xxx-0\n cannot be handled.
> 
> image

such is the case

**Comment by ruanwenjun** [comment]
@stoplyy Should be fixed by #16296

**Pull request merged**